src/trace/trace.jl

```diff
@@ -94,14 +81,10 @@ function forkc(trace :: Trace)
   newtrace = typeof(trace)()
   newtrace.task = Base.copy(trace.task)
   newtrace.spl = trace.spl
-  if trace.spl != nothing
-    gid = trace.spl.alg.group_id
-  else
-    gid = 0
-  end
 
-  n_rand = min(trace.vi.index, length(groupvals(trace.vi, gid, trace.spl)))
-  newtrace.vi = retain(deepcopy(trace.vi), gid, n_rand, trace.spl)
+  n_rand = min(trace.vi.index, length(groupvals(trace.vi, trace.spl)))
+  newtrace.vi = deepcopy(trace.vi)
+  retain!(newtrace.vi, n_rand, trace.spl)
```

Comment:
Here is another problem here: `n_rand` is not essentially the length of `vals` to keep because all variables are flatten (but `newtrace.vi[spl]` is aimed to so if we make `n_rand+1:l` to be variable-level range then it is not consistent).